pkg/webhook/podunavailablebudget/validating/pub_create_update_handler.go

```diff
@@ -0,0 +1,173 @@
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	appsvalidation "k8s.io/kubernetes/pkg/apis/apps/validation"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// PodUnavailableBudgetCreateUpdateHandler handles PodUnavailableBudget
+type PodUnavailableBudgetCreateUpdateHandler struct {
+	// To use the client, you need to do the following:
+	// - uncomment it
+	// - import sigs.k8s.io/controller-runtime/pkg/client
+	// - uncomment the InjectClient method at the bottom of this file.
+	Client client.Client
+
+	// Decoder decodes objects
+	Decoder *admission.Decoder
+}
+
+var _ admission.Handler = &PodUnavailableBudgetCreateUpdateHandler{}
+
+// Handle handles admission requests.
+func (h *PodUnavailableBudgetCreateUpdateHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
+	obj := &policyv1alpha1.PodUnavailableBudget{}
+
+	err := h.Decoder.Decode(req, obj)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+	var old *policyv1alpha1.PodUnavailableBudget
+	//when Operation is update, decode older object
+	if req.AdmissionRequest.Operation == admissionv1beta1.Update {
+		old = new(policyv1alpha1.PodUnavailableBudget)
+		if err := h.Decoder.Decode(
+			admission.Request{AdmissionRequest: admissionv1beta1.AdmissionRequest{Object: req.AdmissionRequest.OldObject}},
+			old); err != nil {
+			return admission.Errored(http.StatusBadRequest, err)
+		}
+	}
+	allErrs := h.validatingSidecarSetFn(obj, old)
+	if len(allErrs) != 0 {
+		return admission.Errored(http.StatusInternalServerError, allErrs.ToAggregate())
+	}
+	return admission.ValidationResponse(true, "")
+}
+
+func (h *PodUnavailableBudgetCreateUpdateHandler) validatingSidecarSetFn(obj, old *policyv1alpha1.PodUnavailableBudget) field.ErrorList {
+	//validate Pub.Spec
+	allErrs := validatePodUnavailableBudgetSpec(obj, field.NewPath("spec"))
+	// when operation is update, validating whether old and new pub conflict
+	if old != nil {
+		allErrs = append(allErrs, validateUpdatePubConflict(obj, old, field.NewPath("spec"))...)
+	}
+	//validate whether pub is in conflict with others
+	pubList := &policyv1alpha1.PodUnavailableBudgetList{}
+	if err := h.Client.List(context.TODO(), pubList, &client.ListOptions{Namespace: obj.Namespace}); err != nil {
+		allErrs = append(allErrs, field.InternalError(field.NewPath(""), fmt.Errorf("query other podUnavailableBudget failed, err: %v", err)))
+	} else {
+		allErrs = append(allErrs, validatePubConflict(obj, pubList.Items, field.NewPath("spec"))...)
+	}
+	return allErrs
+}
+
+func validateUpdatePubConflict(obj, old *policyv1alpha1.PodUnavailableBudget, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if !reflect.DeepEqual(obj.Spec.Selector, old.Spec.Selector) || !reflect.DeepEqual(obj.Spec.TargetReference, old.Spec.TargetReference) {
```

Comment:
shall we check whether the pod scope specified by two selector has some overlapping cases?

Reply:
Is there a scenario where the system owner has created a high-level (project-level) pub protection policy and the owner of a specific module has created a more fine-grained pub, but in fact both pubs do not conflict and can work simultaneously?